node/src/actors/chain_manager/mod.rs

```diff
@@ -1029,8 +1029,113 @@ impl ChainManager {
+                            [&ta_tx.dr_pointer]
+                            .info
+                            .commits
+                            .keys()
+                            .cloned()
+                            .collect();
+
+                        // Reward honest validators
+                        let honest_pkhs: Vec<PublicKeyHash> = commits
+                            .iter()
+                            .filter(|pkh| !ta_tx.out_of_consensus.contains(pkh))
+                            .cloned()
+                            .collect();
+                        for honest_pkh in honest_pkhs {
+                            log::debug!(
+                                "Rewarding {} for {} wit for data request {}",
+                                honest_pkh,
+                                collateral + reward,
+                                ta_tx.dr_pointer
+                            );
+                            let _ = stakes.add_reward(
+                                honest_pkh,
+                                Wit::from(collateral + reward),
+                                block_epoch,
+                            );
+                        }
+
+                        // Refund errored validators
+                        for error_pkh in &ta_tx.error_committers {
+                            log::debug!(
+                                "Refunding {} for {} wit because it revealed an error for data request {}",
+                                error_pkh,
+                                collateral,
+                                ta_tx.dr_pointer
+                            );
+                            let _ =
+                                stakes.add_reward(*error_pkh, Wit::from(collateral), block_epoch);
+                        }
+
+                        // Slash lieing validators
+                        // Collateral was already reserved, so not returning it results in losing it
+                        // Reset the age for witnessing power to 10 epochs in the future
+                        let liar_pkhs: Vec<PublicKeyHash> = ta_tx
+                            .out_of_consensus
+                            .iter()
+                            .filter(|&pkh| !ta_tx.error_committers.contains(pkh))
+                            .cloned()
+                            .collect();
+                        for liar_pkh in &liar_pkhs {
+                            log::debug!(
+                                "Slashing {} for {} wit because it revealed a lie for data request {}",
+                                liar_pkh,
+                                collateral,
+                                ta_tx.dr_pointer
+                            );
+                            let _ = stakes.reset_age(
+                                *liar_pkh,
+                                Capability::Witnessing,
+                                block_epoch + 10,
```

Comment:
Any particular reason for resetting the `coin_age` to -10 instead of just 0 ?

Reply:
I felt it was a bit weird to reset a liar only to `0` given that we also do that for an (honest) committer (albeit at the commit stage), but I don't have a strong opinion on it.